### PR TITLE
OHRM-996: Help Portal

### PR DESCRIPTION
**Pull request by Kaveesha-Silva-OrangeHRM**

### CODE REVIEW

File: src/services/endpoints.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ export const API_ENDPOINT_ATTENDANCE_CONFIGURATION = '/api/v1/time/config';
 // public endpoints
 export const API_ENDPOINT_API_DEFINITION = '/api/v1/api-definition';
 export const API_ENDPOINT_ENABLED_MODULES = '/api/v1/enabled-modules';
-
+export const API_ENDPOINT_HELP = '/api/v1/help/config';
```
Comment (RajithaKumara) [inline src/services/endpoints.ts]: This is not a public endpoint. Update `API_ENDPOINT_HELP` to `API_ENDPOINT_HELP_CONFIG`

File: src/store/help/sagas.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          query: action.payload.query,
+          mode: action.payload.mode,
+        },
+      ),
+    );
+    if (response.data) {
+      yield put(fetchHelpFinished(response.data));
+    } else {
+      yield showSnackMessage(
+        getMessageAlongWithResponseErrors(response, 'Failed to Fetch Help'),
```
Comment (RajithaKumara) [inline src/store/help/sagas.ts]: `Failed to Fetch Help Configurations`

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "orangehrm-opensource",
-  "version": "2.0.0",
+  "version": "2.1.0-alpha1",
```
Comment (RajithaKumara) [inline package.json]: version need to update

File: src/layouts/DrawerContent.tsx [hunk 4]
```diff
@@ -140,11 +181,33 @@ const DrawerContent = (props: DrawerContentProps & DrawerItemListProps) => {
...
+                  style={{marginRight: theme.spacing * -3.5}}
+                />
+              )}
+              {...commonProps}
+            />
+          </>
+        ) : null}
         <Divider />
         <DrawerItem
+          style={{marginLeft: theme.spacing * 0}}
```
Comment (RajithaKumara) [inline src/layouts/DrawerContent.tsx]: Cann't we use just `marginLeft:0`

File: src/layouts/DrawerContent.tsx [hunk 5]
```diff
@@ -140,11 +181,33 @@ const DrawerContent = (props: DrawerContentProps & DrawerItemListProps) => {
         </View>
       </DrawerContentScrollView>
       <View>
+        {isApiCompatible(ORANGEHRM_API_1$3$0) ? (
+          <>
+            <Divider />
+            <DrawerItem
+              style={{marginLeft: theme.spacing * 0}}
```
Comment (RajithaKumara) [inline src/layouts/DrawerContent.tsx]: Cann't we use just `marginLeft:0`

File: src/layouts/DrawerContent.tsx [hunk 6]
```diff
@@ -140,11 +181,33 @@ const DrawerContent = (props: DrawerContentProps & DrawerItemListProps) => {
...
+          </>
+        ) : null}
         <Divider />
         <DrawerItem
+          style={{marginLeft: theme.spacing * 0}}
           label={'Logout'}
           onPress={logoutOnPress}
-          icon={() => <Icon name={'logout'} />}
+          icon={() => (
+            <Icon name={'logout'} style={{marginRight: theme.spacing * -3.5}} />
```
Comment (RajithaKumara) [inline src/layouts/DrawerContent.tsx]: if logout and help hold the same style, you can define icon styles as a const

File: src/screens/login/SelectInstanceHelp.tsx [hunk 7]
```diff
@@ -29,12 +34,24 @@ import MainLayout from 'layouts/MainLayout';
...
-
+import {Button} from 'native-base';
 class SelectInstanceHelp extends React.Component<SelectInstancePropsHelp> {
   onPressClose = () => {
     this.props.navigation.dispatch(StackActions.pop());
   };
 
+  onPressLearnMore = () => {
+    const url =
+      'https://opensourcehelp.orangehrm.com/hc/en-us/articles/360020060020-How-to-Configure-URL-and-Login-to-the-app';
```
Comment (RajithaKumara) [inline src/screens/login/SelectInstanceHelp.tsx]: better to move this URL to the endpoints file or new ts file

File: src/screens/login/SelectInstanceHelp.tsx [hunk 8]
```diff
@@ -29,12 +34,24 @@ import MainLayout from 'layouts/MainLayout';
 import Text from 'components/DefaultText';
 import Icon from 'components/DefaultIcon';
 import withTheme, {WithTheme} from 'lib/hoc/withTheme';
-
+import {Button} from 'native-base';
```
Comment (RajithaKumara) [inline src/screens/login/SelectInstanceHelp.tsx]: Can you use our Button component (`import Button from 'components/DefaultButton';`)  and modify it

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (KaveeshaSilva) [reply package.json]: updated

File: src/screens/login/SelectInstanceHelp.tsx [hunk 9]
```diff
@@ -29,12 +34,30 @@ import MainLayout from 'layouts/MainLayout';
 import Text from 'components/DefaultText';
 import Icon from 'components/DefaultIcon';
 import withTheme, {WithTheme} from 'lib/hoc/withTheme';
-
+import {Button} from 'native-base'; // Cannot use DefaultButton since title always UPPERCASE
+import {SUPPORT_URL} from 'services/help';
+import {SEND_MAIL_SUPPORT_EMAIL_URL} from 'services/help';
```
Comment (RajithaKumara) [inline src/screens/login/SelectInstanceHelp.tsx]:
```
import {SUPPORT_URL} from 'services/help';
import {SEND_MAIL_SUPPORT_EMAIL_URL} from 'services/help';
```
Can't we import as one line?
```
import {SUPPORT_URL, SEND_MAIL_SUPPORT_EMAIL_URL} from 'services/help';
```

File: src/layouts/DrawerContent.tsx [hunk 10]
```diff
@@ -86,6 +100,33 @@ const DrawerContent = (props: DrawerContentProps & DrawerItemListProps) => {
...
+
+  const navigateHelp = () => {
+    let url = helpConfig?.redirectUrls[0]?.url;
+    if (url === undefined) {
+      url = helpConfig?.defaultRedirectUrl;
+    }
+    if (url !== undefined) {
+      Linking.canOpenURL(url).then((supported) => {
+        if (supported) {
+          if (url !== undefined) {
```
Comment (RajithaKumara) [inline src/layouts/DrawerContent.tsx:115]: Do we need to check this again? in line 112 also checked

File: src/layouts/DrawerContent.tsx [hunk 11]
```diff
@@ -140,11 +181,28 @@ const DrawerContent = (props: DrawerContentProps & DrawerItemListProps) => {
         </View>
       </DrawerContentScrollView>
       <View>
+        {isApiCompatible(ORANGEHRM_API_1$3$0) ? (
+          <>
+            <Divider />
+            <DrawerItem
+              style={styles.drawerItem}
+              label={'Help'}
+              onPress={() => {
```
Comment (RajithaKumara) [inline src/layouts/DrawerContent.tsx]: here you can pass prop as `onPress={onPressHelp}`

File: src/store/help/types.ts [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+  url: string;
+}
+
+export interface Help {
+  defaultRedirectUrl: string;
+  redirectUrls: ArticleObject[];
+}
+
+export const FETCH_HELP = 'HELP_FETCH_HELP';
+export const FETCH_HELP_FINISHED = 'HELP_FETCH_HELP_FINISHED';
```
Comment (RajithaKumara) [inline src/store/help/types.ts]: All actions related to fetch config shouldn't be FETCH_HELP_CONFIG kind of names

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/layouts/DrawerContent.tsx (hunk 10 above)
Comment (Kaveesha-Silva-OrangeHRM) [reply src/layouts/DrawerContent.tsx:115]: No need at run time. Did that just for avoiding compile-time error with passing undefined value for OpenURL.

Comment (Kaveesha-Silva-OrangeHRM) [reply src/layouts/DrawerContent.tsx:115]: No need at run time. Did that just for avoiding compile-time error with passing undefined value for OpenURL.

File: src/store/help/actions.ts [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+import {
+  HelpConfig,
+  FETCH_HELP_CONFIG,
+  FETCH_HELP_CONFIG_FINISHED,
+  FetchHelpConfigAction,
+  FetchHelpConfigFinishedAction,
+} from './types';
+import {$PropertyType} from 'utility-types';
+
+export const fetchConigHelp = (
```
Comment (RajithaKumara) [inline src/store/help/actions.ts]: check typo. `fetchConigHelp` => `fetchConfigHelp`

**Review by RajithaKumara — CHANGES_REQUESTED** [review]
Run `node scripts/increment-build-number.js` and update PR

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**